packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -45,7 +45,6 @@ enum DatePickerMode {
   year,
 }
 
-const double _kDatePickerHeaderPortraitHeight = 100.0;
 const double _kDatePickerHeaderLandscapeWidth = 168.0;
```

Comment:
Why are we leaving this one?

Reply:
Neither in tickets nor in my tests I found any issues with displaying this picker in the landscape mode, so I didn't touch this value. However, the problem with overflowing won't appear in that case, as the header text is wrapped to second line if it doesn't have enough space. And that's why (I think so, but didn't check it yet) it will need concrete width to avoid stretching this header to fit whole text.

Reply:
I believe you'll see this problem if you set the text scale factor a bit higher, even in landscape.

Reply:
I'm ok with this PR if it really does fix things for portrait even if we leave landscape with a potential problem though.  Perfect doesn't have to be the enemy of the good here :)